tensorflow/python/ops/image_ops_impl.py

```diff
@@ -2055,7 +2055,8 @@ def _is_png(contents, name=None):
 
 @tf_export('io.decode_image', 'image.decode_image',
            v1=['io.decode_image', 'image.decode_image'])
-def decode_image(contents, channels=None, dtype=dtypes.uint8, name=None):
+def decode_image(contents, channels=None, dtype=dtypes.uint8,
+                 num_returned_dims="3_OR_4", name=None):
```

Comment:
Instead of num_returned_dims can we name the flag something related to its behavior (like, say, expand_animations=True)?

Reply:
Sure, I can make that change. If I'm understanding correctly, the semantics of this flag would be:
* `True` ==> the returned op will produce a 3-D tensor for PNG, JPEG, and BMP and a 4-D tensor for all GIFs, whether animated or not.
* `False` ==> the returned op will produce a 3-D tensor for all file types and will truncate animated GIFs to the first frame.